src/main/java/eu/xenit/gradle/docker/DockerBuildExtension.java

```diff
@@ -51,11 +55,24 @@ public void setTags(List<Object> tags) {
             if (!(tag instanceof String) && !(tag instanceof GString)) {
                 throw new Error("Only strings and gstrings are supported.");
             }
-            converted.add(tag.toString());
+            converted.add(cleanTag(tag.toString()));
```

Comment:
While encoding automatically generated tags might be acceptable, messing around with tags explicitly provided by the user is off-limits.

If the error message when passing an invalid tag is not good or confusing, you can do the check here and throw a better worded exception.

Reply:
Ill move the `cleanTag` invocation to the appropriate block for automatic tags then.